### ISSUE TITLE
get_target_property() called with non-existent target "Sophus::Sophus"

### ISSUE THREAD

**Issue by Nick-Kou** [strasdat/Sophus#310]
A repository called minisam that I had used was built and compiled successfully, with Sophus 8 months ago. After moving over to new PC and attempting to reinstall with the current version of Sophus, I obtain fmt undefined reference errors. For example:
```
[ 60%] Linking CXX executable bundle_adjustment_bal /usr/bin/ld: CMakeFiles/bundle_adjustment_bal.dir/bundle_adjustment_bal.cpp.o: in function Sophus::SO3Base<Sophus::SO3<double, 0> >::normalize() [clone .part.380]':
bundle_adjustment_bal.cpp:(.text+0x1c0): undefined reference to fmt::v6::vprint(fmt::v6::basic_string_view<char>, fmt::v6::format_args)' /usr/bin/ld: CMakeFiles/bundle_adjustment_bal.dir/bundle_adjustment_bal.cpp.o: in function fmt::v6::internal::numeric_specs_checker<fmt::v6::internal::dynamic_specs_handler<fmt::v6::basic_format_parse_context<char, fmt::v6::internal::error_handler> > >::require_numeric_argument() [clone .isra.363] [clone .part.364]':
bundle_adjustment_bal.cpp:(.text.unlikely+0x10): undefined reference to fmt::v6::internal::error_handler::on_error(char const*)' /usr/bin/ld: CMakeFiles/bundle_adjustment_bal.dir/bundle_adjustment_bal.cpp.o: in function unsigned long long fmt::v6::internal::width_checkerfmt::v6::internal::error_handler::operator()<float, 0>(float) [clone .isra.404]':
bundle_adjustment_bal.cpp:(.text.unlikely+0x23): undefined reference to fmt::v6::internal::error_handler::on_error(char const*)' /usr/bin/ld: CMakeFiles/bundle_adjustment_bal.dir/bundle_adjustment_bal.cpp.o: in function unsigned long long fmt::v6::internal::precision_checkerfmt::v6::internal::error_handler::operator()<float, 0>(float) [clone .isra.415]':
bundle_adjustment_bal.cpp:(.text.unlikely+0x37): undefined reference to fmt::v6::internal::error_handler::on_error(char const*)' /usr/bin/ld: CMakeFiles/bundle_adjustment_bal.dir/bundle_adjustment_bal.cpp.o: in function void fmt::v6::print<char [2], , 0>(char const (&) [2])
```

I then decided to use an older version of Sophus and rebuild the minisam repo using a version of Sophus before the inclusion of fmt by doing:
```
git clone https://github.com/strasdat/Sophus.git
git checkout 0c84aed
```
But now I obtain errors building minisam such as:

```
CMake Error at CMakeLists.txt:101 (get_target_property):
get_target_property() called with non-existent target "Sophus::Sophus".

```
The line in 101 CmakeLists.txt is: 
`get_target_property(Sophus_INCLUDE_DIR Sophus::Sophus INTERFACE_INCLUDE_DIRECTORIES)`

It seems as if the Sophus_INCLUDE_DIR variable is empty. Also experienced by others when attempting to compile this minisam repo with an older version of Sophus, dongjing3309/minisam#4

**Comment by sergiud** [comment]
https://github.com/strasdat/Sophus/issues/247#issuecomment-630833938

**Comment by Nick-Kou** [comment]
@sergiud Thanks for the reply. However, after removing the older version and trying the with current version of Sophus for example, 
```
git clone https://github.com/strasdat/Sophus.git 
cd Sophus
mkdir build
cd build
cmake ..
cmake --build .


```


it appears that the issue still persists now. This only occurred after downloading the v1.0 release from here:https://github.com/strasdat/Sophus/releases and then deleting it. It's as if cmake is able to still find the older version even after I made sure that I removed it from /usr/local/include and /usr/local/share/sophus/cmake, as well as in the home directory and downloads. Is there any place I am missing?

**Comment by sergiud** [comment]
When configuring your project, you need to specify where CMake should be looking for Sophus, e.g.,
```bash
$ cmake <build-dir> -DCMAKE_PREFIX_PATH=/path/to/Sophus/build-dir
```
CMake should also emit a warning of the import target `Sophus::Sophus` cannot be found.

**Comment by Nick-Kou** [comment]
The problem was solved. I found there was this folder located in /usr/local/lib/cmake/Sophus
[Sophus.zip](https://github.com/strasdat/Sophus/files/6750335/Sophus.zip)
After removing this file and ensuring there wasn't any other sophus in the locations mentioned before, the variables were not found. I am not sure what affect the above folder had but might have been set after the old installation. Thanks
